### PR TITLE
Region analysis of nameof

### PR DESCRIPTION
**Pull request by jcouv**
Fixes https://github.com/dotnet/roslyn/issues/53591

### CODE REVIEW

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs [hunk 1]
```diff
@@ -35071,7 +35071,7 @@ public static void Main(string[] args)
             var decl = GetOutVarDeclaration(tree, name);
             var refs = GetReferences(tree, name).ToArray();
             Assert.Equal(2, refs.Length);
-            VerifyModelForOutVar(model, decl, refs[0]);
+            VerifyModelForOutVar(model, decl, isDelegateCreation: false, isExecutableCode: false, isShadowed: false, references: refs[0]);
```
Comment (RikkiGibson) [inline src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs:35074]: Could you please clarify what the significance of this change is?

**Comment by RikkiGibson** [comment]
This is a situation where the IDE shows a variable as being captured but it isn't actually lowered into a field on a closure class, so it ends up being misleading for users.

File: src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs (hunk 1 above)
Comment (jcouv) [reply src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs:35074]: The `isExecutableCode: false` is not the default for overload with fewer parameters. Here's what that drives:
```
            if (isExecutableCode)
            {
                Assert.True(dataFlow.Succeeded);
                Assert.True(dataFlow.VariablesDeclared.Contains(symbol, ReferenceEqualityComparer.Instance));

                if (!isDelegateCreation)
                {
                    Assert.True(dataFlow.AlwaysAssigned.Contains(symbol, ReferenceEqualityComparer.Instance));
                    Assert.True(dataFlow.WrittenInside.Contains(symbol, ReferenceEqualityComparer.Instance));

                    var flowsIn = FlowsIn(dataFlowParent, decl, references);
                    Assert.Equal(flowsIn,
                                 dataFlow.DataFlowsIn.Contains(symbol, ReferenceEqualityComparer.Instance));
                    Assert.Equal(flowsIn,
                                 dataFlow.ReadInside.Contains(symbol, ReferenceEqualityComparer.Instance));

                    Assert.Equal(FlowsOut(dataFlowParent, decl, references),
                                 dataFlow.DataFlowsOut.Contains(symbol, ReferenceEqualityComparer.Instance));
                    Assert.Equal(ReadOutside(dataFlowParent, references),
                                 dataFlow.ReadOutside.Contains(symbol, ReferenceEqualityComparer.Instance));

                    Assert.Equal(WrittenOutside(dataFlowParent, references),
                                 dataFlow.WrittenOutside.Contains(symbol, ReferenceEqualityComparer.Instance));
                }
            }
````

**Comment by jcouv** [comment]
@dotnet/roslyn-compiler for review. Small fix. Thanks

**Review by RikkiGibson — APPROVED** [review]

**Comment by 333fred** [comment]
@jcouv I assume we have tests that cover the examples you put in https://github.com/dotnet/roslyn/issues/53591#issuecomment-902076107?

**Comment by jcouv** [comment]
> I assume we have tests that cover the examples you put in #53591 (comment)?

Yes. Those tests existed already in various shapes.

**Review by 333fred — APPROVED** [review]
LGTM (commit 1)

**Pull request merged**